site/src/pages/DeploySettingsPage/ServiceBannerSettingsPage.tsx

```diff
@@ -0,0 +1,245 @@
+    const newBanner = {
+      message: values.message,
+      enabled: values.enabled,
+      background_color: values.backgroundColor,
+    }
+    if (preview) {
+      serviceBannerSend({
+        type: "SET_PREVIEW_BANNER",
+        serviceBanner: newBanner,
+      })
+      return
+    }
+    serviceBannerSend({
+      type: "SET_BANNER",
+      serviceBanner: newBanner,
+    })
+  }
+
+  const initialValues: ServiceBannerFormValues = {
+    message: serviceBanner.message,
+    enabled: serviceBanner.enabled,
+    backgroundColor: serviceBanner.background_color,
+  }
+
+  const form: FormikContextType<ServiceBannerFormValues> =
+    useFormik<ServiceBannerFormValues>({
+      initialValues,
+      validationSchema,
+      onSubmit: (values) => setBanner(values, false),
+    })
+  const getFieldHelpers = getFormHelpers<ServiceBannerFormValues>(form)
+
+  const [backgroundColor, setBackgroundColor] = useState(
+    form.values.backgroundColor,
+  )
+
+  const theme = useTheme()
+
+  return (
+    <>
+      <Helmet>
+        <title>{pageTitle("Service Banner Settings")}</title>
+      </Helmet>
+
+      <Header
+        title="Service Banner"
+        description="Configure a banner that displays a message to all users"
+        docsHref="https://coder.com/docs/coder-oss/latest/admin/high-availability#service-banners"
+      />
+      <Badges>
+        {isEntitled ? <EntitledBadge /> : <DisabledBadge />}
+        <EnterpriseBadge />
+      </Badges>
+
+      {isEntitled ? (
+        <form className={styles.form} onSubmit={form.handleSubmit}>
+          <Stack>
+            <FormControlLabel
+              control={
+                <Switch
+                  {...getFieldHelpers("enabled")}
+                  color="primary"
+                  checked={form.values.enabled}
+                  onChange={() => {
+                    const newState = !form.values.enabled
+                    const newBanner = {
+                      ...form.values,
+                      enabled: newState,
+                    }
+                    setBanner(newBanner, false)
+                    form.setFieldValue("enabled", newState)
+                  }}
+                />
+              }
+              label="Enabled"
+            />
+            <Stack spacing={0}>
+              <TextField
+                {...getFieldHelpers("message")}
+                fullWidth
+                label={Language.messageLabel}
+                variant="outlined"
+                multiline
+                onChange={(e) => {
+                  form.setFieldValue("message", e.target.value)
+                  setBanner(
+                    {
+                      ...form.values,
+                      message: e.target.value,
+                    },
+                    true,
+                  )
+                }}
+              />
+              <FormHelperText>
+                Markdown bold, italics, and links are supported.
+              </FormHelperText>
```

Comment:
Seeing this error in the console on the new banner page:
![Screen Shot 2022-12-05 at 10 19 31 AM](https://user-images.githubusercontent.com/19142439/205673987-22231374-75b6-471d-b94a-a22655f70ef3.png)